packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -38,7 +44,16 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 		// note: don’t claim .astro files with resolveId() — it prevents Vite from transpiling the final JS (import.meta.globEager, etc.)
 		async resolveId(id) {
 			// serve sub-part requests (*?astro) as virtual modules
-			if (parseAstroRequest(id).query.astro) {
+			const { query } = parseAstroRequest(id);
+			if (query.astro) {
+				// Convert /src/pages/index.astro?astro&type=style to /Users/name/
+				// Because this needs to be the id for the Vite CSS plugin to property resolve
+				// relative @imports.
+				if (query.type === 'style' && isBrowserPath(id)) {
+					const outId = npath.posix.join(config.projectRoot.pathname, id);
```

Comment:
just confirming: does this support subpath projects like `/docs`?

Reply:
Yeah, things in `src/` will still start with `/src` in dev mode.

Reply:
Ah no, I meant that if my current root is `.`, but I want to deploy my site to `www.example.com/docs`. In dev, the site should be hosted at `localhost:xxxx/docs`, correct? just confirming that this was still supported.

Reply:
Yes, that's still supported.